src/org/rocketproplab/marginalstability/flightcomputer/hal/LPS22HD.java

```diff
@@ -100,14 +100,16 @@ private void readPressure() {
       i2cDevice.read(REG_PRESSURE_EXTRA_LOW, buffer, 0, 3);
 
       // Perform 2's complement if value is negative
+      byte sign = 0;
       byte mask = (byte)0b10000000;
       if((buffer[2] & mask) > 0){
-          buffer[0] = (byte)(~buffer[0] + 1);
-          buffer[1] = (byte)~buffer[1];
-          buffer[2] = (byte)~buffer[2];
+        sign = 1;
+        buffer[0] = (byte)~buffer[0];
+        buffer[1] = (byte)~buffer[1];
+        buffer[2] = (byte)~buffer[2];
       }
 
-      int rawPressure = (Byte.toUnsignedInt(buffer[2])<<16) + (Byte.toUnsignedInt(buffer[1])<<8) + Byte.toUnsignedInt(buffer[0]);
+      int rawPressure = (Byte.toUnsignedInt(buffer[2])<<16) + (Byte.toUnsignedInt(buffer[1])<<8) + Byte.toUnsignedInt(buffer[0]) + sign;
```

Comment:
I don't think this 2's complement conversion is correct. Try the following test.
```Java
@Test
public void getPressureNegativeMax() {
  MockI2CDevice i2c = new MockI2CDevice();
  BarometerTime time = new BarometerTime();
  LPS22HD barometer = new LPS22HD(i2c, time);
    
  i2c.initValuesThree();
  i2c.readMap.put(0x2A, (byte) -128);
  i2c.readMap.put(0x29, (byte) 0);
  i2c.readMap.put(0x28, (byte) 0);
  barometer.poll();
  assertEquals(-2048, barometer.getPressure(), 0.0005);
}
```